### PR TITLE
ref: Remove event.get_legacy_message()

### PR DESCRIPTION
**Pull request by lynnagara**
- Use event.message everywhere instead of event.get_legacy_message()
- Remove get_legacy_message()

The only remaining usages of this I can find are the Pushover and Pagerduty plugins which are updated here.

### CODE REVIEW

**Comment by fpacifici** [comment]
Looks ok to me, will let the ecosystem chime in on the change in the plugins code

File: src/sentry_plugins/pagerduty/plugin.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def notify_users(self, group, event, fail_silently=False, **kwargs):
         if not self.is_configured(group.project):
             return
 
-        description = event.get_legacy_message()[:1024]
+        description = event.message[:1024]
```
Comment (mitsuhiko) [inline src/sentry_plugins/pagerduty/plugin.py]: Sure this shouldn't rather be `event.title`? `event.message` I would assume is always empty.

File: src/sentry_plugins/pushover/plugin.py [hunk 2]
```diff
@@ -94,7 +94,7 @@ def notify(self, notification):
         title = "%s: %s" % (project.name, group.title)
         link = group.get_absolute_url(params={"referrer": "pushover_plugin"})
 
-        message = event.get_legacy_message()[:256]
+        message = event.message[:256]
```
Comment (mitsuhiko) [inline src/sentry_plugins/pushover/plugin.py]: Same here.

File: tests/sentry_plugins/pushover/test_plugin.py [hunk 3]
```diff
@@ -52,7 +52,7 @@ def test_simple_notification(self):
         request = responses.calls[0].request
         payload = parse_qs(request.body)
         assert payload == {
-            "message": ["{}\n\nTags: level=warning".format(event.get_legacy_message())],
+            "message": ["{}\n\nTags: level=warning".format(event.message)],
```
Comment (mitsuhiko) [inline tests/sentry_plugins/pushover/test_plugin.py]: same here

File: tests/sentry_plugins/pagerduty/test_plugin.py [hunk 4]
```diff
@@ -92,7 +92,7 @@ def test_simple_notification(self):
                 "datetime": event.datetime.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
             },
             "service_key": "abcdef",
-            "description": event.get_legacy_message(),
+            "description": event.message,
```
Comment (mitsuhiko) [inline tests/sentry_plugins/pagerduty/test_plugin.py]: same here.

File: tests/sentry_plugins/pushover/test_plugin.py [hunk 5]
```diff
@@ -89,7 +89,7 @@ def test_emergency_notification(self):
...
```
Comment (mitsuhiko) [inline tests/sentry_plugins/pushover/test_plugin.py]: same here.

**Review by mitsuhiko — DISMISSED** [review]

File: src/sentry_plugins/pagerduty/plugin.py (hunk 1 above)
Comment (lynnagara) [reply src/sentry_plugins/pagerduty/plugin.py]: @mitsuhiko I didn't realise that `event.message` is empty, updating to use title. Should this property still exist then or can it be deleted? https://github.com/getsentry/sentry/blob/master/src/sentry/models/event.py#L539-L543

Comment (mitsuhiko) [reply src/sentry_plugins/pagerduty/plugin.py]: @lynnagara the legacy message did something different. We mostly phased it out now. `event.message` is only filled for event types of type log message. `event.title` returns something for all event types.

Comment (markstory) [reply src/sentry_plugins/pagerduty/plugin.py]: Should we also exclude message from the public schema in discover if it is deprecated/not working well?

**Comment by MeredithAnya** [comment]
@lynnagara @mitsuhiko is there any major change between what the `get_legacy_message` and `title` return? I'm not sure if people are using the alert rule routing right now with the PD plugin but if they are using info in the description it could affect their routing rules. If anything I'm wondering if it's worth sending some sort of warning about that changing - if there is a substantial difference. @manuzope not sure if you have thoughts

**Comment by mitsuhiko** [comment]
@MeredithAnya yes, they are quite different. There are some tests in sentry that show the difference in behavior I believe.

Specifically some SDKs (old ones in particular) can send message + exception in which case `get_legacy_message` would return the message and `title` would be `ExceptionType: exception value`. I know the ruby SDK was only very recently fixed in that regard.

**Comment by manuzope** [comment]
Yes, good catch @MeredithAnya. If users have [event rules set up](https://support.pagerduty.com/docs/global-event-rules#section-events-sent-through-the-api) in PagerDuty, changing the description could break their alert routing. I agree with Meredith. If we're making this change, we should provide an intermediate path for folks and then tell them that we are going to make this change.

Do we need to make this change now or can it wait for a bit while we figure out communication/do due diligence?

**Comment by lynnagara** [comment]
It can wait, can I leave the outreach in the hands of your team? :)

**Comment by manuzope** [comment]
Yes, for sure. Feel free to move forward with pushover though.

Let's also chat offline. I'd like to understand the general urgency of the problem so I know what kind of timeline we are looking at.

**Comment by lynnagara** [comment]
Reverted Pagerduty changes for now

File: src/sentry/models/event.py [hunk 6]
```diff
@@ -120,10 +120,9 @@ def get_interface(self, name):
         return self.interfaces.get(name)
 
     def get_legacy_message(self):
-        # TODO(mitsuhiko): remove this code once it's unused.  It's still
-        # being used by plugin code and once the message rename is through
-        # plugins should instead swithc to the actual message attribute or
-        # this method could return what currently is real_message.
+        # TODO: This is only used in the pagerduty plugin. We should use event.title
+        # ther and remove this function once users have been notified, since PD
```
Comment (markstory) [inline src/sentry/models/event.py]:
```suggestion
        # there and remove this function once users have been notified, since PD
```

**Review by manuzope — APPROVED** [review]

**Pull request merged**